File: buscar_producto.php
<?php
header("Content-Type: application/json");

if (!isset($_GET["codigo"]) || empty($_GET["codigo"])) {
    enviarRespuesta(600, null);    
} else {

    $bd = mysqli_connect("localhost", "root", "", "pos");

    if (!$bd) {
        enviarRespuesta(601, null);    
    }

    $sql = "SELECT producto_nombre, producto_precio, producto_imagen
            FROM productos WHERE producto_codigo = '" . $_GET["codigo"] . "'";

    $res = mysqli_query($bd, $sql);

    if (!$res) {
        enviarRespuesta(602, null);
    }

    if (mysqli_num_rows($res) > 0) {
        while ($filaProducto = mysqli_fetch_assoc($res)) {
            $producto["nombre"] = $filaProducto["producto_nombre"];
            $producto["precio"] = $filaProducto["producto_precio"];
            $producto["imagen"] = $filaProducto["producto_imagen"];
        }
        enviarRespuesta(603, $producto);
    } else {
        enviarRespuesta(604, null);
    }
}

function enviarRespuesta($codigoEstado, $infoProducto) {
    header("HTTP/1.1 " . $codigoEstado);

    $respuesta["estado"] = $codigoEstado;
    $respuesta["nombre"] = empty($infoProducto["nombre"]) ? "" : $infoProducto["nombre"];
    $respuesta["precio"] = empty($infoProducto["precio"]) ? "" : $infoProducto["precio"];
    $respuesta["imagen"] = empty($infoProducto["imagen"]) ? "" : $infoProducto["imagen"];

    echo json_encode($respuesta);
}

File: buscar_producto.py
from flask import Flask, request, jsonify
import mysql.connector

app = Flask(__name__)

@app.route("/producto", methods=["GET"])
def get_producto():
    codigo = request.args.get("codigo")
    if not codigo:
            return jsonify({"status": 400, "producto_nombre": "", "producto_precio": "", "producto_imagen": ""}), 400

    try:
        conn = mysql.connector.connect(host="localhost", user="root", password="", database="pos")
        cursor = conn.cursor(dictionary=True)
        cursor.execute("SELECT producto_nombre, producto_precio, producto_imagen FROM productos WHERE producto_codigo = %s", (codigo,))
        row = cursor.fetchone()
        if row:
            return jsonify({"status": 200, **row}), 200
        else:
            return jsonify({"status": 404, "producto_nombre": "", "producto_precio": "", "producto_imagen": ""}), 404
    except:
        print("ERROR", e)
        return jsonify({"status": 500, "producto_nombre": "", "producto_precio": "", "producto_imagen": ""}), 500

if __name__ == "__main__":
    app.run(debug=True)
